apps/web/pages/api/availability/calendar.ts

```diff
@@ -35,66 +40,153 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     },
   });
   if (!userWithCredentials) {
-    res.status(401).json({ message: "Not authenticated" });
-    return;
+    throw new HttpError({ statusCode: 401, message: "Not authenticated" });
   }
-  const { credentials, ...user } = userWithCredentials;
-
-  if (req.method === "POST") {
-    const { integration, externalId, credentialId } = selectedCalendarSelectSchema.parse(req.body);
-    await prisma.selectedCalendar.upsert({
-      where: {
-        userId_integration_externalId: {
-          userId: user.id,
-          integration,
-          externalId,
-        },
-      },
-      create: {
+  req.userWithCredentials = userWithCredentials;
+  return userWithCredentials;
+}
+
+const watchCalendarSchema = z.object({
```

Comment:
`nitpick`: I'd consider placing this in a reusable file, as I've seen the same schema being used in this PR.

Reply:
It will not be reused anymore